.storybook/recipes/Filters/Filters.tsx

```diff
@@ -1,31 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import FiltersDrawer from '../../../src/components/FiltersDrawer';
 import {
   FiltersPopover,
   FiltersPopoverProps,
 } from '../../../src/components/FiltersPopover/FiltersPopover';
 
+import breakpoint from '../../../src/design-tokens/tier-1-definitions/breakpoints';
 type Props = {
-  /**
-   * Filters component to render.
-   */
-  as?: 'FiltersDrawer' | 'FiltersPopover';
   /**
    * Popover placement options relative to the filters trigger button.
    */
   placement?: FiltersPopoverProps['placement'];
 } & FiltersPopoverProps;
 
 /**
- * Primary UI component for user interaction
+ * Demonstrates usage of both FiltersPopover and FiltersDrawer depending on screensize.
  */
-export const Filters = ({ as, placement, ...other }: Props) => {
+export const Filters = ({ placement, ...other }: Props) => {
+  const [isLarge, setIsLarge] = useState(false);
+  const popoverBreakpoint = parseInt(breakpoint['eds-bp-md']) * 16; /* 1 */
+  useEffect(() => {
+    updateScreenSize();
+    window.addEventListener('resize', updateScreenSize);
+    return () => {
+      window.removeEventListener('resize', updateScreenSize);
+    };
+  });
+
+  const updateScreenSize = () => {
+    if (window.innerWidth >= popoverBreakpoint) {
+      setIsLarge(true);
+    } else {
+      setIsLarge(false);
+    }
+  };
   return (
     <div>
-      {as === 'FiltersDrawer' && <FiltersDrawer {...other} />}
-      {as === 'FiltersPopover' && (
-        <FiltersPopover placement={placement} {...other} />
-      )}
+      {!isLarge && <FiltersDrawer {...other} />}
+      {isLarge && <FiltersPopover placement={placement} {...other} />}
```

Comment:
Do you think it's possible to have this kind of functionality built into a wrapper component or will devs need to add this themselves every time?

Reply:
I think it would be possible, but I want to leave it up to devs since when to swap between Drawer and Popover can depend on content size or other conditions. Basically want to punt state management as much as possible lol